libs/common/src/middleware/feature-flags/validate-feature-flag-enabled.ts

```diff
@@ -0,0 +1,50 @@
+import { RequestHandler } from 'express';
+import { HttpStatusCode } from 'axios';
+import { isPortalFacilityAmendmentsFeatureFlagEnabled, isTfmDealCancellationFeatureFlagEnabled } from '../../helpers';
+import { FeatureFlag } from '../../helpers/is-feature-flag-enabled';
+
+/**
+ * Creates a middleware function to check if the given feature flag is enabled
+ *
+ * This will send status `404` if the feature flag is disabled and call `next` otherwise
+ */
+const generateBackendFeatureFlagMiddleware =
+  (featureFlagName: FeatureFlag, isFeatureFlagEnabled: () => boolean): RequestHandler =>
+  (req, res, next) => {
+    if (!isFeatureFlagEnabled()) {
+      console.info(`Feature flag ${featureFlagName} disabled, accessing ${req.originalUrl}`);
+      return res.sendStatus(HttpStatusCode.NotFound);
+    }
+
+    return next();
+  };
+
+/**
+ * Creates a middleware function to check if the given feature flag is enabled
+ *
+ * This will redirect to `/not-found` if the feature flag is disabled and call `next` otherwise
+ */
+export const generateFrontendFeatureFlagMiddleware =
+  (featureFlagName: FeatureFlag, isFeatureFlagEnabled: () => boolean): RequestHandler =>
+  (req, res, next) => {
+    if (!isFeatureFlagEnabled()) {
+      console.info(`Feature flag ${featureFlagName} disabled, accessing ${req.originalUrl}`);
```

Comment:
Can we add "whilst" or "when" please:

```suggestion
      console.info(`Feature flag ${featureFlagName} disabled whilst accessing ${req.originalUrl}`);
```